pkg/controller/federatedcluster/controller.go

```diff
@@ -38,32 +38,44 @@ import (
 	"github.com/golang/glog"
 )
 
+// ClusterController is responsible for maintaining the status of each
+// FederatedCluster in a particular namespace.
 type ClusterController struct {
-	fedClient  fedclientset.Interface
+	// fedClient is used to access Federation resources in the host cluster.
+	fedClient fedclientset.Interface
+	// kubeClient is used to access Secrets in the host cluster.
 	kubeClient kubeclientset.Interface
-	crClient   crclientset.Interface
+	// crClient is used to access the cluster registry in the host cluster.
+	crClient crclientset.Interface
 
 	// clusterMonitorPeriod is the period for updating status of cluster
 	clusterMonitorPeriod time.Duration
 
-	mu              sync.RWMutex
+	mu sync.RWMutex
+
+	// knownClusterSet is the set of clusters known to this controller.
 	knownClusterSet sets.String
-	// clusterClusterStatusMap is a mapping of clusterName and cluster status of last sampling
+
+	// clusterClusterStatusMap is a mapping of clusterName and cluster status of
```

Comment:
(No action required) clusterClusterStatus?  I just noticed how ridiculous this name is.

Reply:
:)